src/CoreWCF.WebHttp/src/CoreWCF/Description/WebHttpBehavior.cs

```diff
@@ -0,0 +1,1356 @@
+                    }
+                }
+            }
+        }
+
+        void ValidateNoOperationHasEncodedXmlSerializer(OperationDescription od)
+        {
+            XmlSerializerOperationBehavior xsob = ((KeyedByTypeCollection<IOperationBehavior>)od.OperationBehaviors).Find<XmlSerializerOperationBehavior>();
+            if (xsob != null && (xsob.XmlSerializerFormatAttribute.Style == OperationFormatStyle.Rpc || xsob.XmlSerializerFormatAttribute.Use == OperationFormatUse.Encoded))
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.RpcEncodedNotSupportedForNoneMessageVersion, od.Name, od.DeclaringContract.Name, od.DeclaringContract.Namespace)));
+            }
+        }
+
+        void ValidateNoBareMessageContractWithMultipleParts(MessageDescription md, string opName, string contractName)
+        {
+            if (IsTypedMessage(md) && md.Body.WrapperName == null)
+            {
+                if (md.Body.Parts.Count > 1)
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(
+                        SR.Format(SR.InvalidMessageContractWithoutWrapperName, opName, contractName, md.MessageType)));
+                }
+
+                if (md.Body.Parts.Count == 1 && md.Body.Parts[0].Multiple)
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.MCAtMostOneRequestBodyParameterAllowedForUnwrappedMessages, opName, contractName, md.MessageType)));
+                }
+            }
+        }
+
+        void ValidateNoMessageContractHeaders(MessageDescription md, string opName, string contractName)
+        {
+            if (md.Headers.Count != 0)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(
+                    SR.Format(SR.InvalidMethodWithSOAPHeaders, opName, contractName)));
+            }
+        }
+
+        internal class MessagePassthroughFormatter : IClientMessageFormatter, IDispatchMessageFormatter
+        {
+            public object DeserializeReply(Message message, object[] parameters)
+            {
+                return message;
+            }
+
+            public void DeserializeRequest(Message message, object[] parameters)
+            {
+                parameters[0] = message;
+            }
+
+            public Message SerializeReply(MessageVersion messageVersion, object[] parameters, object result)
+            {
+                return result as Message;
+            }
+
+            public Message SerializeRequest(MessageVersion messageVersion, object[] parameters)
+            {
+                return parameters[0] as Message;
+            }
+        }
+
+        //static internal JavascriptCallbackResponseMessageProperty TrySetupJavascriptCallback(string callbackParameterName)
+        //{
```

Comment:
You can remove this

Reply:
This was addressed in b54a2808ec6f4245ccf662c73a2b441ea6c85814.